docs/architecture/adr-020-deterministic-square-construction.md

```diff
@@ -0,0 +1,82 @@
+
+Earlier designs were incorporated around the notion of interaction between the block proposer and the transaction submitter. A user that wanted to submit a PFB would go to a potential block proposer, provide them with the transaction, the proposer would then reserve a position in the square for the transaction and finally the transaction submitter would sign the transaction with the provided share index. However, Celestia may have 100 potential block proposers which are often hidden from the network. Furthermore, tranasctions often reach a block proposer through a gossip network, severing the ability for the block proposer to directly communicate with the transaction submitter. Lastly, new transactions with greater fees might arrive causing the block proposer to want to shuffle the transactions around in the square. The response to these problems was to come up with "non-interactive defaults" (first mentioned in [ADR006](./adr-006-non-interactive-defaults.md)).
+
+## Decision
+
+Block proposers should only have the freedom to decide transaction inclusion and ordering (Note there are some existing constraints on ordering). Interaction is still permitted (in the sense that a transaction submitter can directly communicate with a block proposer) but not directly supported by the Celestia network. The non-interactive defaults become mandatory.
+
+Square construction is thus to be reduced to the simple deterministic function:
+
+```go
+func ConstructSquare(txs []Tx) []Share
+```
+
+and it's couterpart
+
+```go
+func DeconstructSquare(shares []Share) []Tx
+```
+
+Block proposers need not know nor care about the internals of square construction. With gradual improvements to construction techniques that best incorporate the tradeoffs between square size (impacting bandwidth and storage) with proof size (impacting light clients), the latest culmination in [ADR013](./adr-013-non-interactive-default-rules-for-zero-padding.md), there is little room for validators to implement optimizations. Such existing control to block proposers also means they are able to spam the network; building full size squares that include a single PFB. Lastly, a deterministic construction mechanism has the benefit of a more compact representation of the data that can freely move between different representations (this also supports a simpler compact blocks protocol):
+
+```go
+[]Tx <=> []Share <=> EDS
+```
+
+Validation in `ProcessProposal` is simplified to reconstructing the square given the transaction set and then evaluating whether the computed data root matches the proposed one. 
```

Comment:
[not blocking comment]

mind elaborating on this difference a bit more and what is being simplified? are they doing something else atm? does this new approach remove the necessity for any other checks?

Reply:
very related to https://github.com/celestiaorg/celestia-app/pull/1633#discussion_r1171347793, can just respond there

Reply:
Yes, for example, you don't need to check that the blobs are ordered by namespace. You will still need to validate the `BlobTxs` 

[Will write this down]